chia/full_node/block_store.py

```diff
@@ -338,14 +340,27 @@ async def get_blocks_by_hash(self, header_hashes: List[bytes32]) -> List[FullBlo
         return ret
 
     async def get_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
-        cursor = await self.db.execute(
-            "SELECT block from block_records WHERE header_hash=?",
-            (self.maybe_to_hex(header_hash),),
-        )
-        row = await cursor.fetchone()
-        await cursor.close()
-        if row is not None:
-            return BlockRecord.from_bytes(row[0])
+
+        if self.db_wrapper.db_version == 2:
+
+            cursor = await self.db.execute(
+                "SELECT block_record FROM full_blocks WHERE header_hash=?",
+                (header_hash,),
+            )
+            row = await cursor.fetchone()
+            await cursor.close()
```

Comment:
If we want to explicitly close the cursor it should probably be done with the context manager.  Otherwise, intermediate exceptions hazard the cursor not being closed.

There are other cases, I won't point them each out.

```suggestion
            async with self.db.execute(
                "SELECT block_record FROM full_blocks WHERE header_hash=?",
                (header_hash,),
            ) as cursor:
                row = await cursor.fetchone()
```

Reply:
I agree. would you mind if I submit a separate PR for this? it's an unrelated change and affects more places than this patch update